src/ui/public/timefilter/time_history.js

```diff
@@ -0,0 +1,33 @@
+import moment from 'moment';
+import { PersistedLog } from 'ui/persisted_log';
+import { TIME_MODES } from 'ui/timepicker/modes';
+
+class TimeHistory {
+  constructor() {
+    const historyOptions = {
+      maxLength: 10,
+      filterDuplicates: true
+    };
+    this.history = new PersistedLog('kibana.timepicker.timeHistory', historyOptions);
+  }
+
+  add(time) {
+    if (!time || time.mode === TIME_MODES.QUICK) {
```

Comment:
I'd argue it makes sense to include quick times in the history for the sake of consistency. When I first pulled down the PR and started playing with it I started switching between Quick options and I thought something was broken because they weren't showing up in the Recent list. I think a user might want to be able to flip back and forth between a couple quick options without having to find it in the giant Quick list each time. Might just be me though, if others disagree I won't be insistent.

Reply:
That makes sense. I will remove this check